src/index.ts

```diff
@@ -23,19 +23,53 @@ const defaultApps = [
   require('./plugins/default')
 ]
 
-export const createProbot = (options: Options) => {
-  options.webhookPath = options.webhookPath || '/'
-  options.secret = options.secret || 'development'
+export class Probot {
+  public server: express.Application
+  public webhook: any
+  public logger: Logger
+
+  private options: Options
+  private robots: Robot[]
+  private app: () => string
```

Comment:
Does this need to be `private`? If I wanted to add some middleware for example, I would do:

```js
probot.app.use(myMiddleware)
```

Would this still work in a TS world?

Reply:
This is a horribly named variable. This is the GitHub App function that returns a jwt. It's exposed on `Robot`. `server` is the express app.

Reply:
ack very true. Although kinda the same question - you'd still be able to use this method outside of the `Probot` instance? I can see it being useful maybe somehow?

Reply:
This caught my eye as well, but I think if you wanted to use your own `app`, you'd just pull in `createApp` to get your own App outside of the Probot instance. Since it's already imported here, we can add `createApp` to the exports at the bottom of this file to enable that. 

Then in your own project, you could do this:

```typescript
import {createApp, createRobot} from 'probot'

const app = createApp({
    id: process.env.APP_ID,
    cert: process.env.PRIVATE_KEY
})

const robot = createRobot({app})
...
```

Thoughts?